src/OnboardingSPA/components/ThemeContextProvider/index.js

```diff
@@ -1,28 +1,54 @@
 import { useState, useEffect, createContext } from '@wordpress/element';
+import { useSelect } from '@wordpress/data';
+import { store as nfdOnboardingStore } from '../../store';
+import { setFlow } from '../../utils/api/flow';
 import { THEME_DARK, THEME_LIGHT } from '../../../constants';
 
 const ThemeContext = createContext();
 
-function getPreferredColorScheme() {
+function usePreferredColorScheme() {
+	const { currentData } = useSelect( ( select ) => {
+		return {
+			currentData:
+				select( nfdOnboardingStore ).getCurrentOnboardingData(),
+		};
+	} );
+
+	if ( currentData.sitegenThemeMode ) {
+		return currentData.sitegenThemeMode;
+	}
+
 	if (
 		window.matchMedia &&
 		window.matchMedia( '(prefers-color-scheme: dark)' ).matches
 	) {
 		return THEME_DARK;
 	}
+
 	return THEME_LIGHT;
 }
 
 const ThemeProvider = ( { children } ) => {
-	const [ theme, setTheme ] = useState( getPreferredColorScheme );
+	const preferredColorScheme = usePreferredColorScheme();
+	const [ theme, setTheme ] = useState( preferredColorScheme );
+
+	const { currentData } = useSelect( ( select ) => {
+		return {
+			currentData:
+				select( nfdOnboardingStore ).getCurrentOnboardingData(),
+		};
+	} );
 
 	useEffect( () => {
 		const colorSchemeMediaQuery = window.matchMedia(
 			'(prefers-color-scheme: dark)'
 		);
 
 		const handleChange = ( event ) => {
-			setTheme( event.matches ? THEME_DARK : THEME_LIGHT );
+			const newTheme = event.matches ? THEME_DARK : THEME_LIGHT;
+			setTheme( newTheme );
+			currentData.sitegenThemeMode = newTheme;
+			setFlow( currentData );
```

Comment:
Considering we are changing the data shouldn't we also do `setCurrentOnboardingData` and sync it to the store?